sys/include/net/lorawan/hdr.h

```diff
@@ -0,0 +1,266 @@
+ *          The structure of the header is as follows:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.unparsed}
+ *                      1                   2                   3
+ *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-
+ * |Mtype| RFU |Maj|           LoRaWAN Address                   ...
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * ...             | Frame Control |         Frame Counter         |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * @see <a href="https://www.lora-alliance.org/portals/0/specs/LoRaWAN%20Specification%201R0.pdf">
+ *          LoRaWAN spcefication, section 4
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    /**
+     * @brief message type and major version
+     *
+     * @details the message type are the 3 most significant bytes and the
+     * major version are the 2 less significant bytes. There are 3 bytes
+     * in the middle reserved for future usage.
+     * This module provides helper functions to set and get:
+     * * lorawan_hdr_set_mtype()
+     * * lorawan_hdr_get_mtype()
+     * * lorawan_hdr_set_maj()
+     * * lorawan_hdr_get_maj()
+     */
+    uint8_t mt_maj;
+    le_uint32_t addr;                           /**< 32 bit LoRaWAN address */
+    uint8_t fctrl;                              /**< frame control */
+    le_uint16_t fcnt;                           /**< frame counter */
+} lorawan_hdr_t;
+
+/**
+ * @brief Join request packet representation
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t mt_maj;                             /**< mtype and major version holder */
+    le_uint64_t app_eui;                        /**< application EUI */
+    le_uint64_t dev_eui;                        /**< device EUI */
+    le_uint16_t dev_nonce;                      /**< device nonce */
+    le_uint32_t mic;                            /**< message integrity code */
+} lorawan_join_request_t;
+
+/**
+ * @brief Join accept packet representation
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t mt_maj;                             /**< mtype and major version holder */
+    uint8_t app_nonce[LORAMAC_APP_NONCE_LEN];   /**< application nonce */
+    uint8_t net_id[LORAMAC_NETWORK_ID_LEN];     /**< network id */
+    uint8_t dev_addr[LORAMAC_DEVADDR_LEN];      /**< device address */
+    uint8_t dl_settings;                        /**< downlink settings */
+    uint8_t rx_delay;                           /**< first reception window delay */
+} lorawan_join_accept_t;
+
+/**
+ * @brief Set LoRaWAN header MType
+ *
+ * @param[in] hdr LoRaWAN hdr
+ * @param[in] mtype desired MType
+ */
+static inline void lorawan_hdr_set_mtype(lorawan_hdr_t *hdr, uint8_t mtype)
+{
+    hdr->mt_maj &= 0x1f;
```

Comment:
I think those hard coded values should be provided as defines. What about `LORAWAN_HDR_MTYPE_MASK`, `LORAWAN_HDR_MTYPE_POS` or something ? The same could apply with other fields in the header.

Reply:
agreed

Reply:
done